tests/integration_test.py

```diff
@@ -179,22 +186,44 @@ def wait_until(f, timeout_secs=10):
 
 
 def test_that_pulsevideosrc_recovers_if_pulsevideo_crashes(
-        pulsevideo_via_activation):
+        pulsevideo_via_activation, tmpdir):
+
+    cmd = shquote(pulsevideo_cmdline())
+    with open("%s/pulsevideo" % tmpdir, 'w') as f:
```

Comment:
Right, so here we overwrite `tmpdir/pulsevideo` which was created earlier by the `pulsevideo_via_activation`. It works because the `tmpdir` fixture that's injected here is the same directory as the `tmpdir` injected into `pulsevideo_via_activation`. Just to be paranoid, have you checked the logs to see that pulsevideo is actually crashing?

Reply:
Yes it is.  It might be worth adding a check in the test for the fault injection logging.

Reply:
I've added the check that the failures are actually happening.